runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -236,7 +254,29 @@ int count() {
      */
     @Nonnull
     Collection<CorfuRecord<V, M>> scanAndFilter(@Nonnull final Predicate<CorfuRecord<V, M>> p) {
-        return new ArrayList<>(corfuTable.scanAndFilter(p));
+        return corfuTable.scanAndFilter(p);
+    }
+
+    /**
+     * Scan and filter by entry.
+     *
+     * @param entryPredicate Predicate to filter the entries.
+     * @return Collection of filtered entries.
+     */
+    @Nonnull
+    List<CorfuStoreEntry<K, V, M>> scanAndFilterByEntry(
+            @Nonnull final Predicate<CorfuStoreEntry<K, V, M>> entryPredicate) {
+        return corfuTable.scanAndFilterByEntry(recordEntry ->
+                entryPredicate.test(new CorfuStoreEntry<>(
+                        recordEntry.getKey(),
+                        recordEntry.getValue().getPayload(),
+                        recordEntry.getValue().getMetadata())))
+                .parallelStream()
```

Comment:
Just make sure that we do not get in trouble with parallelStream as it uses common fork join pool.

Reply:
These are not running long-running tasks but just wrapping the entry. Also, this `parallelStream` does not run the user's predicate and hence is bounded.